### DISCUSSION TITLE
New Release: May 17, 2022 🎊

### DISCUSSION THREAD

**Discussion by lastzero** [photoprism/photoprism#2328] [General]
This update features search UX improvements, a new Docker base image based on Debian 12 "Bookworm", as well as fixes for recently discovered issues. Front- and backend [translations](https://translate.photoprism.app/) in numerous languages have been added and updated. Thanks to all involved!

#### What is new?

- File Formats: [Added native support for WebP images](https://github.com/photoprism/photoprism/issues/1226) and [animated GIFs](https://github.com/photoprism/photoprism/commit/82d61d1f93961daa5b9ffea31e2ae4ab8897d2a7)
- UX: [RAW files are skipped by default when downloading photos and ZIP archives](https://github.com/photoprism/photoprism/issues/2234)
- Auth: [Passwords must have at least 8 characters to mitigate brute-force attacks](https://github.com/photoprism/photoprism/issues/2248)
- Search: [User interface and database performance optimizations](https://github.com/photoprism/photoprism/issues/1438)
- Search: [Fixed occasional lag when entering queries in the search toolbar](https://github.com/photoprism/photoprism/issues/1995)
- Search: [Improved `album:...` filter supports numeric names, `albums:..` also AND/OR](https://github.com/photoprism/photoprism/issues/1994)
- Search: [Improved `camera:...` and `lens:...` filters accept names in addition to IDs](https://github.com/photoprism/photoprism/issues/2079)
- Search: [Added `square:yes` and `landscape:yes` filters](https://github.com/photoprism/photoprism/issues/2169)
- Albums: ["Add to album" dialog preloads more names for auto-completion](https://github.com/photoprism/photoprism/pull/2152)
- Albums: [Album names are shortened if necessary to avoid errors when saving](https://github.com/photoprism/photoprism/issues/2181)
- Albums: [Fixed accidental creation of duplicates by pressing Enter multiple times](https://github.com/photoprism/photoprism/issues/2233)
- People: [Improved logging and fixed potential issues with matching unrecognized faces](https://github.com/photoprism/photoprism/issues/2182)
- Places: [Number of pictures rendered on the map has been limited to 500,000](https://github.com/photoprism/photoprism/commit/49e923232380117c8b1eab9ff5b41de878d46ab2)
- Library: [Added button to clear log history in *Library* > *Errors*](https://github.com/photoprism/photoprism/discussions/1683)
- Library: [RAW previews and the number of actual files are shown under Originals](https://github.com/photoprism/photoprism/issues/2273)
- Library: [Disabled hidden files warning while indexing as it can be misleading](https://github.com/photoprism/photoprism/issues/2189)
- Index: [Fixed errors when re-indexing libraries with archived photos](https://github.com/photoprism/photoprism/issues/2257)
- Index: [RAW and video conversion commands run in a virtual home directory](https://github.com/photoprism/photoprism/issues/2262)
- Thumbnails: [Reduced default JPEG quality from 92 to 85 to optimize storage and loading](https://github.com/photoprism/photoprism/issues/2215)
- Metadata: [Manual local time changes are always preserved when reindexing](https://github.com/photoprism/photoprism/issues/2239)
- Metadata: [Fixed Exif orientation flag when converting HEIF/HEIC images to JPEG](https://github.com/photoprism/photoprism/discussions/2214)
- Metadata: [Fault-tolerant parsing of timestamps from Exif and JSON sidecar files](https://github.com/photoprism/photoprism/issues/625)
- Metadata: [Improved parsing of two-digit years in original file paths](https://github.com/photoprism/photoprism/issues/2271)
- Metadata: [Exif IFD1 tags with existing IFD0 values are ignored to improve standard compliance](https://github.com/photoprism/photoprism/issues/2231)
- Metadata: [Brute-force search is skipped by default if no Exif headers were found in JPEG, PNG, TIFF, and HEIF files](https://github.com/photoprism/photoprism/issues/2196)
- Metadata: [SubSecDateTimeOriginal and SubSecCreateDate timestamps are preferred](https://github.com/photoprism/photoprism/issues/2320)
- WebDAV: [Up- and download sync can no longer be enabled at the same time to prevent unexpected behavior](https://github.com/photoprism/photoprism/issues/1785)
- WebDAV: [Added timeout/retry settings and improved handling of sync errors](https://github.com/photoprism/photoprism/issues/1781)
- WebDAV: [Fixed sharing videos and uploading automatically created albums](https://github.com/photoprism/photoprism/issues/2293)
- CLI: [Renamed `--config-file` to `--defaults-yaml` and improved command help](https://github.com/photoprism/photoprism/issues/2250)
- CLI: [Added short names for common config flags, e.g. `-i` for `--wakeup-interval`](https://github.com/photoprism/photoprism/issues/2195)
- CLI: [Run `photoprism show tags` to display metadata tags and supported standards](https://github.com/photoprism/photoprism/issues/2252)
- CLI: [Run `photoprism show formats` to display supported media and sidecar file formats](https://github.com/photoprism/photoprism/issues/2247)
- CLI: [Run `photoprism show filters` to display a search filter overview with examples](https://github.com/photoprism/photoprism/commit/7291c1d70329d85af2dfc1e9de512d28378974a5)
- Config: [Improved FFmpeg parameters for Intel QSV hardware transcoding](https://github.com/photoprism/photoprism/issues/2222)
- Config: [Added NVIDIA hardware video transcoding support for sponsors](https://github.com/photoprism/photoprism/issues/2125)
- Config: [Added `--disable-raw` flag to disable indexing and conversion of RAW files](https://github.com/photoprism/photoprism/issues/2227)
- Config: [Added `--resolution-limit` option to skip high-resolution images when indexing](https://github.com/photoprism/photoprism/issues/1017)
- Docker: [New Debian 12 "Bookworm" base image with FFmpeg 4.4.1 and Darktable 3.8.1](https://github.com/photoprism/photoprism/issues/2178)
- Docker: [Added default users and groups for enhanced video transcoding compatibility](https://github.com/photoprism/photoprism/issues/2228)
- Translations: [Added Swedish, Romanian, Turkish, Lithuanian, Bulgarian, Malay, and Croatian](https://translate.photoprism.app/)

<hr /><em>This discussion was created from the release <a href='https://github.com/photoprism/photoprism/releases/tag/220517-b9c68f8f'>May 17, 2022</a>.</em>

**Comment by aniqueta** [comment]
@lastzero Great update! 

Unfortunately, my instance no longer works with this update, and I'm stuck troubleshooting. The log (with debug/trace) shows that PhotoPrism is starting normally. My old session works, and I do not need to log in again. However, the loading bar takes a long time and then ultimately no photos load. Likewise, all settings are grayed out. There are no additional entries created in the log. 

With a private browser window, the login screen comes up, and but it just loads indefinitely after entering username and password. Unfortunately, the logs shows nothing about what is not working. It does add a log item about a new session being created though. 

Is this happening for anyone else? Any ideas what this "hanging" behavior without log entries could mean? Thanks. 

Edit: Watching the log, background tasks seems to be working and operating on the library correctly, e.g., routine check, facial recognition, moments analysis, etc.

And here's the debug log for the login, which seems to show a timeout after 4 minutes, but not enough information on why?

``` 
DEBU[2022-05-17T12:31:57-07:00] http: GET / (200) [4.556278ms]               
DEBU[2022-05-17T12:31:57-07:00] http: GET /static/build/app.69775a0d495d6171eecb.css (200) [567.678µs] 
DEBU[2022-05-17T12:31:57-07:00] http: GET /manifest.json?1652785431 (200) [2.401332ms] 
DEBU[2022-05-17T12:31:57-07:00] http: GET /static/build/app.1bfa8af39f30449f9e5b.js (200) [89.970708ms] 
DEBU[2022-05-17T12:31:57-07:00] http: GET /static/icons/logo.svg (200) [192.392µs] 
DEBU[2022-05-17T12:31:57-07:00] http: GET /static/build/2d8017489da689caedc1.woff2 (200) [356.27µs] 
DEBU[2022-05-17T12:31:58-07:00] http: GET /sw.js (200) [205.65µs]            
DEBU[2022-05-17T12:32:08-07:00] session: created                             
DEBU[2022-05-17T12:35:05-07:00] http: GET / (200) [3.106244ms]               
DEBU[2022-05-17T12:36:17-07:00] http: POST /api/v1/session (200) [4m9.912334012s] 
DEBU[2022-05-17T12:36:18-07:00] api: abort /api/v1/config with code 401 (Please log in and try again) 
DEBU[2022-05-17T12:36:18-07:00] http: GET /api/v1/config (401) [114.623µs]   
DEBU[2022-05-17T12:36:19-07:00] http: GET / (200) [3.341163ms]               
DEBU[2022-05-17T12:36:19-07:00] http: GET /api/v1/ws (200) [10m0.146014087s] 
DEBU[2022-05-17T12:36:19-07:00] http: GET /static/build/app.69775a0d495d6171eecb.css (200) [979.226µs] 
DEBU[2022-05-17T12:36:19-07:00] http: GET /manifest.json?1652785431 (200) [2.41764ms] 
DEBU[2022-05-17T12:36:19-07:00] http: GET /static/build/app.1bfa8af39f30449f9e5b.js (200) [280.633947ms] 
DEBU[2022-05-17T12:36:19-07:00] http: GET /static/icons/logo.svg (200) [203.901µs] 
DEBU[2022-05-17T12:36:19-07:00] http: GET /static/build/2d8017489da689caedc1.woff2 (200) [341.683µs] 
DEBU[2022-05-17T12:36:20-07:00] http: GET /sw.js (200) [189.763µs]           
```

**Reply by kvalev** [reply]
Try updating the user password using the CLI: `docker-compose exec photoprism photoprism users update --help`

**Reply by aniqueta** [reply]
@kvalev Thanks, but that did not change this behavior. (I tried it even though my password was already complex and greater than eight characters per change log above.)

**Reply by lastzero** [reply]
Are you using the default admin user or any custom setup? The long waiting time can be a protection against brute force attacks, which isn't very sophisticated right now. Try resetting the failure counts in the users table if you know a little SQL.Also make sure your password matches, you have no scripts that automatically access the UI / API in the background, and the password has at least 8 characters.

**Reply by aniqueta** [reply]
Thanks @lastzero! 

Yes, default user is admin.

Looking at the SQL database, login_attempts is already 0, so I'm not sure what more resetting can do… 

The password matches, since if I use the wrong password, I get the appropriate "Invalid credentials" error immediately. When using the correct password, the log shows "session: created" immediately (but then the web UI hangs as noted).

Password has been complex and greater than eight characters. I also tested changing the password from CLI to something simpler, if say special characters were the problem. No change to the issue. 

The only thing custom is my instance is compiled from source and run on macOS. It is not using Docker. Last version few versions ran flawlessly. I've also checked if there's weird macOS permissions issues, by running as root and with full disk access. Results in the same behavior with no new information in the logs. 

Appreciate any other ideas. Thanks again.

**Reply by lastzero** [reply]
We have not tested it natively on macOS yet, so there MAY be compatibility issues. In particular, the new build scripts set explicit LD library paths. Those are likely not compatible with operating systems other than Linux. 

How do you configure PhotoPrism? Please check if you are using the correct CLI parameters as some changed. This is documented in the release notes. In particular those related to config files and defaults:

- https://github.com/photoprism/photoprism/issues/2250

**Reply by aniqueta** [reply]
I think my build is working correctly, since I get a working binary, the log shows that background operations are working, and the WebUI does load, but I can audit that further. To run, I'm using `--config-path` and from the log the correct configuration is loading.

**Reply by lastzero** [reply]
Could it be a proxy, browser or other frontend issue?

**Reply by aniqueta** [reply]
I've tried multiple browsers, devices, etc, and nothing has changed in client configuration since the update. :/ Is there any way to get more logging out of PhotoPrism? It seems weird that there's nothing showing up in the logs even with debug/trace? Thanks for the engagement on this!

**Reply by lastzero** [reply]
It might actually help to reduce the log level in case there is a feedback loop when a log message is triggered about failing to send a log message to the UI....

**Reply by lastzero** [reply]
Repeated log messages should now be omitted and a minimum level for the frontend viewer has been added: 
- https://github.com/photoprism/photoprism/issues/2335

If you want, you can try out if your issues have improved with our latest preview:
- https://docs.photoprism.app/release-notes/#development-preview

**Reply by aniqueta** [reply]
@lastzero Figured out what happened. The MariaDB database had become corrupt. Repairing the database solved the issue. The build above did not produce any new errors though to help troubleshoot. I just took a guess to check the database integrity. Perhaps there can be some error logging when the database doesn't respond properly? Thanks.

**Reply by lastzero** [reply]
Thanks for the update! If you automatically pull/install new mariadb versions, make sure to enable the auto update functionality so you're not getting stuck with an outdated schema.

**Reply by lastzero** [reply]
FYI: I successfully built and ran a macOS (Intel) binary yesterday. To actually being able to run an instance and index files, I had to update the TensorFlow model download scripts in our repo as they have been using absolute paths for a few commands. We had issues with the Docker PATH variable in the past and this was a workaround that got permanent. Surprisingly, nobody ever complained.

**Reply by aniqueta** [reply]
> FYI: I successfully built and ran a macOS (Intel) binary yesterday. To actually being able to run an instance and index files, I had to update the TensorFlow model download scripts in our repo as they have been using absolute paths for a few commands. We had issues with the Docker PATH variable in the past and this was a workaround that got permanent. Surprisingly, nobody ever complained.

Yes! It builds pretty seamlessly. :) [Edit: Disregard. Looks like the download scripts have been updated in develop to be more compatible. Thank you!]

> Thanks for the update! If you automatically pull/install new mariadb versions, make sure to enable the auto update functionality so you're not getting stuck with an outdated schema.

I'm not following. MariaDB updates automatically using Homebrew, for me. Do I need to do something more to keep the DB schema up-to-date via PhotoPrism? I couldn't find anything in the docs on this. Thanks.

**Reply by lastzero** [reply]
> Yes! It builds pretty seamlessly. :) [Edit: Disregard. Looks like the download scripts have been updated in develop to be more compatible. Thank you!]

Yep! Was going to ask you to try again :)

Let us know when something breaks in the future, or send a PR to make the scripts compatible with macOS **and** Linux. Ideally FreeBSD too if you have experience with that.

> I'm not following. MariaDB updates automatically using Homebrew, for me. Do I need to do something more to keep the DB schema up-to-date via PhotoPrism? I couldn't find anything in the docs on this. Thanks.

Our migrations are automated and should not break unless you are using MySQL 8 or a very old, unsupported MariaDB version.

However, MariaDB itself sometimes requires a migration after upgrading (not always, just sometimes):

- https://docs.photoprism.app/getting-started/troubleshooting/mariadb/#version-upgrade
- https://github.com/photoprism/photoprism/blob/develop/docker/examples/docker-compose.yml#L128

Hope this helps!

**Reply by aniqueta** [reply]
Let me share what I'm doing, and then please tell me if this matches what you did/recommend, and if a PR from me would be worthwhile. I'm an amateur. :) 

What I've done while I experiment is create a custom install script the pulls from the repo and then concatenates the below to the Makefile. The below comments out things that aren't needed on macOS, as far as I'm aware, and then I build using `make all install-macos DEST_DIR=/opt/photoprism`. So, this isn't a build command that works across all platforms at once, but it does create a path to build on macOS without disrupting anything for Linux/Docker. 

```
install-macos:
	$(info Installing in "$(DESTDIR)"…)
#	On macOS, DESTDIR typically already exists, especially if we're updating in place
#	@[ ! -d "$(DESTDIR)" ] || (echo "ERROR: Install path '$(DESTDIR)' already exists!"; exit 1)
	mkdir --mode=$(INSTALL_MODE) -p $(DESTDIR)
#	On macOS, brew install libtensorflow@1 and export LIBRARY_PATH="/usr/local/opt/libtensorflow@1/lib"
#	env TMPDIR="$(BUILD_PATH)" ./scripts/dist/install-tensorflow.sh $(DESTDIR)
	rm -rf --preserve-root $(DESTDIR)/include
	(cd $(DESTDIR) && mkdir -p bin sbin lib assets config config/examples)
	./scripts/build.sh prod "$(DESTDIR)/bin/$(BINARY_NAME)"
#	On macOS, app runs as logged in user/admin, root, or custom daemon user, and this appears to only for Docker
#	[ -f "$(GOBIN)/gosu" ] || go install github.com/tianon/gosu@latest
#	cp $(GOBIN)/gosu $(DESTDIR)/sbin/gosu
	[ ! -f "$(GOBIN)/exif-read-tool" ] || cp $(GOBIN)/exif-read-tool $(DESTDIR)/bin/exif-read-tool
	rsync -r -l --safe-links --exclude-from=assets/.buildignore --chmod=a+r,u+rw ./assets/ $(DESTDIR)/assets
	wget -O $(DESTDIR)/assets/static/img/wallpaper/welcome.jpg https://cdn.photoprism.app/wallpaper/welcome.jpg
	cp scripts/dist/heif-convert.sh $(DESTDIR)/bin/heif-convert
	cp internal/config/testdata/*.yml $(DESTDIR)/config/examples
#	On macOS/source build, changing ownership of the large storage directory when upgrading build is slow and not needed 
#	chown -R $(INSTALL_USER) $(DESTDIR)
	chown -R $(INSTALL_USER) $(DESTDIR)/bin $(DESTDIR)/lib
#	chmod -R $(INSTALL_MODE) $(DESTDIR)
	chmod -R $(INSTALL_MODE_BIN) $(DESTDIR)/bin $(DESTDIR)/lib
	@echo "PhotoPrism $(BUILD_TAG) has been successfully installed in \"$(DESTDIR)\".\nEnjoy!"
```

The other piece is to setup the macOS environment beforehand, installing dependencies from Homebrew and setting additional environment variables. I think this is better scripted separately from the Makefile?

``` 
brew install coreutils grep libtensorflow@1 node
PATH="/usr/local/opt/coreutils/libexec/gnubin:/usr/local/opt/grep/libexec/gnubin:$PATH"
# For some reason the Go build doesn't work on macOS unless this is set using export
export LIBRARY_PATH="/usr/local/opt/libtensorflow@1/lib"
```

Right now, I have a pretty hacky custom script that does install and automatic updates, as well as manages permissions, since I run this under a separate daemon user as a system daemon. 

If this seems generally right and low maintenance, then I'm happy to send through the PR and also help update the docs for others.

**Reply by xyb** [reply]
After some experimentation, I've finally found the correct setup to build PhotoPrism with the latest libtensorflow on macOS using Homebrew. Here are the commands that worked for me:
```sh
❯ brew install libtensorflow

❯ pkg-config tensorflow --cflags --libs
-I/opt/homebrew/Cellar/libtensorflow/2.19.0/include/tensorflow -L/opt/homebrew/Cellar/libtensorflow/2.19.0/lib -ltensorflow -ltensorflow_framework

❯ export C_INCLUDE_PATH=/opt/homebrew/Cellar/libtensorflow/2.19.0/include/
❯ export LIBRARY_PATH=/opt/homebrew/Cellar/libtensorflow/2.19.0/lib

❯ make build-go
```

**Reply by lastzero** [reply]
Make sure to also build the frontend, so you can use the UI :)

**Comment by bossanovaorca** [comment]
after upgrading to this release I can no longer reach the PhotoPrism UI and the Docker container log doesn't seem to show any errors.  I'm not sure how to debug any further though, because the container ends up taking up so much of my server's resources that I have to shut down the container to be able to do anything else!

I ran into this problem with the previous Docker `preview` image so I rolled back to the stable branch, but now the new stable image has this issue as well

**Reply by lastzero** [reply]
Maybe chmod or other commands are running on Startup and this is why you need to wait or disable this as documented and shown in the logs?

**Reply by bossanovaorca** [reply]
yeah I guess it was just taking a lot longer. I just let it run for hours and it finally started up

**Reply by lastzero** [reply]
Note you can disable running chmod/chown on Startup. See log messages to learn more.

**Comment by gpulido** [comment]
Hello!
I'm having a strange problem with the update, it was working without any issues with the latest tag, but after this release came out I'm facing a photoprims/storage permissions problem and the container doesn´t start:
```
started 220517-bookworm as uid 1000 (amd64-prod)
init: updating filesystem permissions
PHOTOPRISM_DISABLE_CHOWN="true" disables permission updates
Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
https://docs.photoprism.app/getting-started/troubleshooting/
file umask....: "0002" (u=rwx,g=rwx,o=rx)
home directory: /photoprism
assets path...: /opt/photoprism/assets
storage path..: /photoprism/storage
config path...: default
cache path....: default
backup path...: /photoprism/storage/backups
import path...: /photoprism/import
originals path: /photoprism/originals
running as uid 1000
/opt/photoprism/bin/photoprism start
Storage folder "/photoprism/storage" is not writable, see https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions
```
I have the storage mounted to a host local volume and it has all read and write permissions, I have even make a chmod a+rw to the entire folder, also I have changed the ownership to be sure and the uid/gid of the container to match the same of the volume, without any luck.
Before the release it was working with the same folders and volumes, also If I roll back to the March version it starts to  work again.
The docker host is a linux and I have been using photoprism almost since the begining.
I don't know what to check more.

**Reply by lastzero** [reply]
Maybe the UID setting was previously ignored and the container was running as root? Also use a+rwX to allow all users to write, not just a+rw.

**Reply by krystiancharubin** [reply]
facing the same issue with latest version, even set permission wide open and still getting the same error
```
started 220517-bookworm as root (amd64-prod)
Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
https://docs.photoprism.app/getting-started/troubleshooting/
file umask....: "0002" (u=rwx,g=rwx,o=rx)
home directory: /photoprism
assets path...: /opt/photoprism/assets
storage path..: /photoprism/storage
config path...: default
cache path....: default
backup path...: /photoprism/storage/backups
import path...: /photoprism/import
originals path: /photoprism/originals
switching to uid 1100:1100
/opt/photoprism/bin/photoprism start
Storage folder "/photoprism/storage" is not writable, see https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions
```
```
rancher@rancheros ~/docker/stack/photo $ id
uid=1100(rancher) gid=1100(rancher) groups=1100(rancher),1101(docker),1103(sudo)
rancher@rancheros ~/docker/stack/photo $ ls -alh photoprism/
total 5272
drwxrwxrwx    7 rancher  rancher        9 May 17 03:39 .
drwxr-xr-x    6 rancher  docker         8 May 18 00:22 ..
drwxrwxrwx    7 rancher  rancher        7 Apr 26 07:22 albums
drwxrwxrwx    2 rancher  rancher        3 Apr 25 17:42 backup
drwxrwxrwx    4 rancher  rancher        5 Apr 25 17:44 cache
drwxrwxrwx    2 rancher  rancher        4 Apr 25 17:42 config
-rwxrwxrwx    1 rancher  rancher     8.2M Apr 25 23:36 index.db
-rwxrwxrwx    1 rancher  rancher       16 Apr 25 17:42 serial
drwxrwxrwx    5 rancher  rancher      958 May  9 04:46 sidecar
```

EDIT:
220517-bullseye seems to be working

**Reply by lastzero** [reply]
Might be because the UID 1100 doesn't exist inside the Docker image. Would It be an option to run it as UID 1000 for example? Do you manually set the home directory using an env variable?

**Reply by gpulido** [reply]
Hello after using 220517-bullseye and a couple of restarts it worked. Latest is still not working though.

**Reply by lastzero** [reply]
That's why I suspect it is related to the Docker image, in particular users, the init script or home directory. It would be good to know if it works with UID 0 or 1000, and also if you have set any custom environment variables.

**Reply by krystiancharubin** [reply]
I am running in RancherOS which does not let you create users so stuck witch rancher(1100) user and passing in the ids using PHOTOPRISM_UID
PHOTOPRISM_GID
In linuxserver when they create the default abs user the set the ids based on PUID and PGID environment variables if they are passed in. Maybe we can do the same thing here.

**Reply by lastzero** [reply]
Can you please open an issue for RancherOS compatibility? 👍

**Reply by gpulido** [reply]
> That's why I suspect it is related to the Docker image, in particular users, the init script or home directory. It would be good to know if it works with UID 0 or 1000, and also if you have set any custom environment variables.

Sorry for the late replay, I'm running it using docker swarm it is being deployed as root but using the 1000:1000 user:group inside the compose, the storage folder has a+rwX and the owner for all the files is the 1000:10000
The variables that are being used:
```
HOTOPRISM_SITE_URL=**edited
PHOTOPRISM_SITE_TITLE=PhotoPrism
PHOTOPRISM_CAPTION=Browse your life
PHOTOPRISM_DESCRIPTION=Personal Photo Management powered by Go and Google TensorFlow. Free and open-source.
PHOTOPRISM_AUTHOR=**edited
PHOTOPRISM_UPLOAD_NSFW=true
PHOTOPRISM_HIDE_NSFW=false
PHOTOPRISM_EXPERIMENTAL=false
PHOTOPRISM_DEBUG=true
PHOTOPRISM_READONLY=false
PHOTOPRISM_PUBLIC=false
PHOTOPRISM_ADMIN_PASSWORD=**edited
PHOTOPRISM_WEBDAV_PASSWORD=**edited
PHOTOPRISM_TIDB_HOST=0.0.0.0
PHOTOPRISM_TIDB_PORT=2343
PHOTOPRISM_TIDB_PASSWORD=**edited
PHOTOPRISM_DATABASE_DRIVER=mysql
PHOTOPRISM_DATABASE_DSN=photoprism:photoprism@tcp(db:3306)/photoprism?parseTime=true
PHOTOPRISM_SIDECAR_HIDDEN=false
PHOTOPRISM_THUMB_FILTER=lanczos
PHOTOPRISM_THUMB_UNCACHED=false
PHOTOPRISM_THUMB_SIZE=2048
MYSQL_ROOT_PASSWORD=**edited
MYSQL_USER=**edited
MYSQL_PASSWORD=**edited
MYSQL_DATABASE=**edited
````
and my compose file:
```
version: '3.5'

services:
	app:
    image: photoprism/photoprism:latest
    env_file: /var/data/config/photoprism/photoprism.env
    user: "1000:1000"
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    networks:
      - internal
      - traefik_public
    working_dir: "/photoprism"
    volumes:
     - /etc/localtime:/etc/localtime:ro
     - /originalsfolder:/photoprism/originals
     - /importfolder:/photoprism/import
     - /exportfolder:/photoprism/export
     - /var/data/photoprism/storage:/photoprism/storage

    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.photoprism.entrypoints=https"
        - "traefik.http.routers.photoprism.rule=Host(`photoprism.example.com`)"
        - "traefik.http.routers.photoprism.tls.certresolver=namecheap"
        - "traefik.http.services.photoprism.loadbalancer.server.port=2342"
		
	db:
    image: mariadb:10.5
    security_opt:
      - seccomp:unconfined
      - apparmor:unconfined
    env_file: /var/data/config/photoprism/photoprism.env
    command: |
      --character-set-server=utf8mb4
      --collation-server=utf8mb4_unicode_ci
      --max-connections=1024
    networks:
      - internal
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/data/runtime/photoprism/db:/var/lib/mysql

networks:
  traefik_public:
    external: true
  internal:
    driver: overlay
    ipam:
      config:
        - subnet: 172.16.90.0/24
```
Hope it helps, If you need me to test anything please let me know.

**Reply by lastzero** [reply]
There is a new preview build available that should work with UID 1100.

**Reply by gpulido** [reply]
Using the preview tag and changing the user to 1100:1100 (and with the storage folder with rwx in all files for all groups) and still doesn't work, same error:
```
started 220519 as uid 1100 (amd64-prod)
init: updating filesystem permissions
PHOTOPRISM_DISABLE_CHOWN="true" disables permission updates
Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
https://docs.photoprism.app/getting-started/troubleshooting/
file umask....: "0002" (u=rwx,g=rwx,o=rx)
home directory: /photoprism
assets path...: /opt/photoprism/assets
storage path..: /photoprism/storage
config path...: default
cache path....: default
backup path...: /photoprism/storage/backups
import path...: /photoprism/import
originals path: /photoprism/originals
running as uid 1100
/opt/photoprism/bin/photoprism start
Storage folder "/photoprism/storage" is not writable, see https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions
```

**Reply by lastzero** [reply]
Did you also change permissions for dot files in the storage folder? Can you show the output of `ls -alh `?

**Reply by gpulido** [reply]
Hello, it seems that there are not dotfiles on my storage, at least at the first level:

```
/var/data/photoprism/storage# ls -alh
total 12K
drwxrwxrwx  7 1100 gpt  8 may 20 00:02 .
drwxrwxrwx  6 gpt  gpt  6 may 18 01:25 ..
drwxrwxrwx  7 1100 gpt  7 ene 29  2021 albums
drwxrwxrwx  2 1100 gpt  2 feb 19 19:32 backup
drwxrwxrwx  6 1100 gpt  7 may 18 13:22 cache
-rwxrwxrwx  1 1100 gpt 16 dic  5  2020 serial
drwxrwxrwx  3 1100 gpt  7 mar 24 00:40 settings
drwxrwxrwx 21 1100 gpt 21 ene 16 20:13 sidecar
```
As you see the user has been changed to 1100 by the container when running

**Reply by lastzero** [reply]
Can you start the container directly as UID 1100 without using the environment variables (which are for starting as root and change the uid in the init script)?

**Reply by lastzero** [reply]
And also disable chmod during startup?

**Reply by gpulido** [reply]
I have removed the user: 1100:1100 from the compose and added the PHOTOPRISM_DISABLE_CHOWN="true" to the env.
```
started 220519 as root (amd64-prod)
Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
https://docs.photoprism.app/getting-started/troubleshooting/
file umask....: "0002" (u=rwx,g=rwx,o=rx)
home directory: /photoprism
assets path...: /opt/photoprism/assets
storage path..: /photoprism/storage
config path...: default
cache path....: default
backup path...: /photoprism/storage/backups
import path...: /photoprism/import
originals path: /photoprism/originals
running as uid 0
/opt/photoprism/bin/photoprism start
Storage folder "/photoprism/storage" is not writable, see https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions
```

**Reply by lastzero** [reply]
This is the line of code causing the error. Do you see anything wrong with it?

https://github.com/photoprism/photoprism/blob/develop/scripts/dist/audit.sh#L17

**Reply by gpulido** [reply]
Hello,
I have been made some test and is a problem with the debian bookworm version.
My test:
I pulled and run an interative sesion to a bare debian:bookworm:
```
sudo docker run -i -t debian:bookworm bash
```
Then just move to the home and create the audit.sh file with the following data:
```
#!/bin/bash

STORAGE_PATH=${PHOTOPRISM_STORAGE_PATH:-storage}

set -e

# create directory if not exists
mkdir -p "${STORAGE_PATH}" || (echo "Failed creating storage folder \"$STORAGE_PATH\", see $DOC_URL" 1>&2; exit 1)

# check directory permissions
[[ -w "${STORAGE_PATH}" ]] || \
  (echo "Storage folder \"$STORAGE_PATH\" is not writable, see $DOC_URL" 1>&2; exit 1)

# create and delete test file
(touch "${STORAGE_PATH}/is-writable" 2>/dev/null && rm "${STORAGE_PATH}/is-writable") || \
  (echo "Failed creating test file in storage folder, see $DOC_URL" 1>&2; exit 1)
```
Then chmod a+x audit.sh and run it

It fails with the permission error.

However if I made the exact same test inside a bullseye that I also fetch with:

```
sudo docker run -i -t debian:bullseye bash
```

It works without problem:
```
root@37d540302cd8:/home#chmod a+x audit.sh
root@37d540302cd8:/home# ./audit.sh
root@37d540302cd8:/home# ls
audit.sh  storage
root@37d540302cd8:/home#

```

**Reply by lastzero** [reply]
We have removed the incorrect permission check on startup:
- https://github.com/photoprism/photoprism/issues/2334

Feel free to test if our latest preview works for you:
- https://docs.photoprism.app/release-notes/#development-preview

**Reply by gpulido** [reply]
It is working now, thank you!!

**Comment by lastzero** [comment]
When upgrading, pease make sure to update your docker-compose.yml based on our example. 

https://dl.photoprism.app/docker/docker-compose.yml
